### ISSUE TITLE
[Common Lisp] Implement new Concept Exercise: assignment

### ISSUE THREAD

**Issue by TheLostLambda** [exercism/v3#1765]
This issue describes how to implement the `assignment` concept exercise for the Common Lisp track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Learning objectives

- Be able to update simple dynamic and lexical variables using `setf`
- Be able to update simple simple "places" (like `car` or `elt`) using `setf`
- Be able to update more complex and / or nested places (`subseq`, `(car (cdr ...))`, etc) using `setf`
- Assigning several places at once (sequential, left to right)
- Know some examples of "places" that can be updated by `setf`
- Know that `setf` evaluates to the newly assigned value
- Be aware of the shorthand `incf` and `decf` operators and their operation
- Understand the "place-swapping"`shiftf` and `rotatef` macros

## Out of scope

- The historical `setq` assignment operator
- The `set` operator (exclusive to dynamic / special variables)
- Hash tables (this exercise is a prerequisite for `hash-tables`)
- The rare `psetf` variant
- The definition of new "places" or "modifiers" (`defsetf`, `define-modify-macro`, etc)

## Concepts

- `assignment`

## Prerequisites

- `variables`
- `basic-lists`
- `strings`
- `expressions`

## Resources to refer to

- [Generalized Variables](https://www.cs.cmu.edu/Groups/AI/html/cltl/clm/node80.html)

### Hints

- [Practical Common Lisp](http://www.gigamonkeys.com/book/variables.html)

### After

Generic setters in Common Lisp are somewhat reminiscent of "lenses" in functional programming, perhaps it would be worth drawing some comparisons and highlighting the differences between the concepts.

We should definitely show off the extensibility of the "places" mechanism and how new places and modifiers can be defined using things like `defsetf` and `define-modify-macro`.

## Representer

Nothing yet

## Analyzer

We might want to check that certain constructs are being used, like `rotatef` instead of a series of `setf` calls with a temporary value.

## Implementing

Start simple with just updating variables with `setf`, then several at a time, then introduce basic places with lists and / or strings, before showing off a trickier update (like a nested place or subsequence) so the benefits of `setf` are made clear.

Calculating things like the fibonacci sequence involve keeping track of the last two values and "shifting" them along. Using `shiftf` makes for a really elegant, one-line implementation:

```lisp
(loop :repeat 10 :with a = 0 :and b = 1 :collect (shiftf a b (+ a b)))
; => (0 1 1 2 3 5 8 13 21 34)
```

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.
